RecoMET/METFilters/plugins/ChargedHadronTrackResolutionFilter.cc

```diff
@@ -0,0 +1,149 @@
+
+
+
+// system include files
+#include <memory>
+#include <iostream>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDFilter.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidateFwd.h"
+#include "DataFormats/METReco/interface/PFMET.h"
+#include "DataFormats/METReco/interface/PFMETFwd.h"
+#include "DataFormats/METReco/interface/PFMETCollection.h"
+//
+// class declaration
+//
+
+class ChargedHadronTrackResolutionFilter : public edm::EDFilter {
```

Comment:
should be thread-aware
better be edm::global::EDFilter
but may be faster to say edm::stream